src/Analysis/GWASV2/Shared/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx

```diff
@@ -80,65 +75,80 @@ const AttritionTableRow = ({
+    };
     setAfr(getSizeByColumn('AFR'));
     setAsn(getSizeByColumn('ASN'));
     setEur(getSizeByColumn('EUR'));
     setHis(getSizeByColumn('HIS'));
   }, [breakdownColumns]);
 
-  const determineChartIcon = (rowTypeInput) => {
-    if (rowTypeInput === 'Cohort') {
-      return null;
+  const determineChartIcon = () => {
+    if (rowType === 'Covariate' || rowType === 'Outcome') {
+      if (rowObject.variable_type === 'concept') {
+        return <BarChart />;
+      }
+      return <EulerDiagram />;
     }
+    return null;
+  };
 
-    /*
-       TODO: Write logic such that if the covariate is numeric, it is bar chart
-       and if the covariate is dichotomous, it should be a euler diagram
-    */
-    return (Math.random() > 0.5) ? <BarChart /> : <EulerDiagram />;
+  const rowName = () => {
+    if (rowType === 'Outcome') {
+      return outcome.variable_type === 'concept'
+        ? outcome.concept_name
+        : 'this is for case control, TODO';
+    }
+    if (rowType === 'Covariate') {
+      return rowObject.concept_name
+        ? rowObject.concept_name
+        : rowObject.provided_name;
+    }
+    // For Cohort Logic
+    return selectedCohort.cohort_name;
   };
+
   return (
     <tr>
-      <td className='gwasv2-attrition-table--leftpad'>
-        {rowType}
-      </td>
+      <td className='gwasv2-attrition-table--leftpad'>{rowType}</td>
       <td className='gwasv2-attrition-table--chart'>
         {determineChartIcon(rowType)}
       </td>
-      <td>{rowName}</td>
-      <td
-        className='gwasv2-attrition-table--rightborder'
-      >
+      <td>{rowName()}</td>
+      <td className='gwasv2-attrition-table--rightborder'>
         {status === 'loading' ? <Spin size='small' /> : breakdownSize || 0}
       </td>
       <td className='gwasv2-attrition-table--leftpad'>
         {status === 'loading' ? <Spin size='small' /> : afr || 0}
       </td>
-      <td>
-        {status === 'loading' ? <Spin size='small' /> : asn || 0}
-      </td>
-      <td>
-        {status === 'loading' ? <Spin size='small' /> : eur || 0}
-      </td>
-      <td>
-        {status === 'loading' ? <Spin size='small' /> : his || 0}
-      </td>
+      <td>{status === 'loading' ? <Spin size='small' /> : asn || 0}</td>
+      <td>{status === 'loading' ? <Spin size='small' /> : eur || 0}</td>
+      <td>{status === 'loading' ? <Spin size='small' /> : his || 0}</td>
     </tr>
   );
 };
 
 AttritionTableRow.propTypes = {
-  cohortDefinitionId: PropTypes.number,
   otherCohortDefinitionId: PropTypes.number,
   rowType: PropTypes.string.isRequired,
-  rowName: PropTypes.string.isRequired,
   covariateSubset: PropTypes.array.isRequired,
+  outcome: PropTypes.object,
+  rowObject: PropTypes.object.isRequired,
+  selectedCohort: PropTypes.object,
   sourceId: PropTypes.number.isRequired,
 };
 
 AttritionTableRow.defaultProps = {
-  cohortDefinitionId: undefined,
   otherCohortDefinitionId: undefined,
+  selectedCohort: undefined,
+  outcome: undefined,
```

Comment:
yes 👍

Reply:
Updated to not required and outcome: null,